app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -191,18 +210,71 @@ void addOrders(List<Order> orders) {
 
         /** Returns the column that contains the given instant, creating it if it doesn't exist. */
         Column getColumnContainingTime(ReadableInstant instant) {
-            LocalDate date = new DateTime(instant).toLocalDate();  // a day in the local time zone
-            DateTime start = date.toDateTimeAtStartOfDay();
+            Column returnColumn;
+            switch (mChartColumnType) {
+                case Column.ENCOUNTER:
+                    returnColumn =  getEncounterColumn(instant);
+                    break;
+                case Column.TIMED:
+                    returnColumn =  getTimedColumn(instant);
+                    break;
+                case Column.DAILY:
+                default:
+                    returnColumn =  getDailyColumn(instant);
+                    break;
+            }
+            return returnColumn;
+        }
+
+        Column getDailyColumn(ReadableInstant instant) {
+            LocalDateTime date = new DateTime(instant).toLocalDateTime();  // a day in the local time zone
+            DateTime start = date.withTime(0, 0, 0, 0).toDateTime();
+            DateTime end = date.millisOfDay().withMaximumValue().toDateTime();
             long startMillis = start.getMillis();
-            if (!mColumnsByStartMillis.containsKey(startMillis)) {
-                int admitDay = Utils.dayNumberSince(mAdmissionDate, date);
+            long endMillis = end.getMillis();
+            Interval columnInterval = new Interval(startMillis, endMillis);
+
+            if (!mColumnsByStartMillis.containsKey(columnInterval)) {
+                int admitDay = Utils.dayNumberSince(mAdmissionDate, date.toLocalDate());
                 String admitDayLabel = (admitDay >= 1) ?
                     mResources.getString(R.string.day_n, admitDay) : "–";
                 String dateLabel = date.toString("d MMM");
-                mColumnsByStartMillis.put(startMillis, new Column(
-                    start, start.plusDays(1), admitDayLabel + "<br>" + dateLabel));
+                mColumnsByStartMillis.put(columnInterval, new Column(
+                    columnInterval, admitDayLabel + "<br/>" + dateLabel));
+            }
+            return mColumnsByStartMillis.get(columnInterval);
+        }
+
+        Column getEncounterColumn(ReadableInstant instant) {
```

Comment:
For `column_type: 'encounter'`, I can't work out what the `column_time` parameter actually does. To me, it looks like it doesn't make a difference to the way that the columns are computed, and each encounter will have a new column created for it. Could you clarify your design for me a bit please?